exo-deploy/src/deployers/aws-deployer.ls

```diff
@@ -47,12 +52,44 @@ class AwsDeployer
 
 
   nuke: ->
-    @terraform
-      ..get (err) ->
+    @terraform-file-builder.generate-provider-credentials!
+    process.stdout.write "terraform starting nuke from AWS"
+    @terraform.destroy (err) ~>
         | err => return process.stdout.write err.message
-        process.stdout.write "terraform starting nuke from AWS"
-        ..destroy (err) ~>
-          | err => return process.stdout.write err.message
+        @_remove-hosted-zone @domain-name
+
+
+  _get-hosted-zone-id: (done) ->
+    @_hosted-zone-exists @domain-name, (id) ~>
+      if id then done id
+      else @_create-hosted-zone @domain-name, (id) -> done id
+
+
+  _hosted-zone-exists: (domain-name, done) ->
+    @route53 = new Aws.Route53 {api-version: '2013-04-01'}
+      ..list-hosted-zones null, (err, data) ~>
+        | err => return process.stdout.write err.message
+        for hosted-zone in data.HostedZones
+          if hosted-zone.Name is "#{domain-name}." then return done hosted-zone.Id
+        return done no
+
+
+  _create-hosted-zone: (domain-name, done) ->
+    params =
+      CallerReference: uuid.v4!
+      Name: domain-name
+    @route53.create-hosted-zone params, (err, data) ~>
+      | err => return process.stdout.write err.message
```

Comment:
Here as well. If `_create-hosted-zone`'s callee is expecting a callback with the result only, won't calling `done process.stdout.write err.message` provide an error message as the result of the `_create-hosted-zone` function call? Or is that just poor design in the first place? Should `_create-hosted-zone`'s callee expect a callback with two parameters `(result, err)`?